src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java

```diff
@@ -468,4 +469,64 @@ void testVscVoltageControlWithOneSideDisconnected() {
         assertActivePowerEquals(0, network.getVscConverterStation("cs2").getTerminal());
         assertVoltageEquals(vcs2, network.getVscConverterStation("cs2").getTerminal().getBusView().getBus());
     }
+
+    @Test
+    void testAcEmuWithOperationalLimits() {
+        Network network = HvdcNetworkFactory.createHvdcLinkedByTwoLinesAndSwitch(HvdcConverterStation.HvdcType.VSC);
+        // without limit p=195
+        network.getHvdcLine("hvdc23")
+                .newExtension(HvdcOperatorActivePowerRangeAdder.class)
+                .withOprFromCS2toCS1(180)
+                .withOprFromCS1toCS2(170)
+                .add();
+
+        LoadFlow.Runner loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        LoadFlowParameters p = new LoadFlowParameters();
+        p.setHvdcAcEmulation(true);
+        LoadFlowResult result = loadFlowRunner.run(network, p);
+
+        assertTrue(result.isFullyConverged());
+
+        // Active flow capped at limit. Output has losses (due to VSC stations)
+        assertEquals(170, network.getHvdcConverterStation("cs2").getTerminal().getP(), DELTA_POWER);
```

Comment:
Can you used dedicated method for voltage or power assertions `assertActivePowerEquals` and `assertVoltageEquals`?

Reply:
Done